CMakeLists.txt

```diff
@@ -352,7 +352,7 @@ if(CMAKE_C_COMPILER_ID STREQUAL "GNU" OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
   add_definitions(-D_GNU_SOURCE)
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND CMAKE_SIZEOF_VOID_P EQUAL 8)
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64" AND CMAKE_SIZEOF_VOID_P EQUAL 8 AND NOT PREFER_LUA)
```

Comment:
It seems this isn't needed at all for LuaJIT >= 2.1.0 beta3.  That would be a better check than adding the architecture.

Reply:
Changed the check to look for `LUAJIT_VERSION LESS "2.1.0-beta3".